nodes/list_mutators/polygon_sort.py

```diff
@@ -0,0 +1,289 @@
+            return
+
+        self.last_mode = self.mode
+        self.update_sockets(context)
+        updateNode(self, context)
+
+    direction : StringProperty(
+        name="Direction", default="X", update=update_xyz_direction)
+
+    mode : EnumProperty(
+        name="Mode", items=modeItems, default="D", update=update_mode)
+
+    last_mode : EnumProperty(
+        name="Last Mode", items=modeItems, default="D")
+
+    point_P : FloatVectorProperty(
+        name="Point P", description="Reference point for distance and angle calculation",
+        size=3, default=(1, 0, 0), update=updateNode)
+
+    point_D : FloatVectorProperty(
+        name="Direction", description="Reference direction for projection and angle calculation",
+        size=3, default=(1, 0, 0), update=updateNode)
+
+    descending : BoolProperty(
+        name="Descending", description="Sort in the descending order",
+        default=False, update=updateNode)
+
+    def sv_init(self, context):
+        self.inputs.new('SvVerticesSocket', "Verts")
+        self.inputs.new('SvStringsSocket', "Polys")
+        self.inputs.new('SvVerticesSocket', "Direction").prop_name = "point_D"
+        self.outputs.new('SvVerticesSocket', "Vertices")
+        self.outputs.new('SvStringsSocket', "Polygons")
+        self.outputs.new('SvStringsSocket', "Indices")
+        self.outputs.new('SvStringsSocket', "Distances")
+
+    def draw_buttons(self, context, layout):
+        col = layout.column(align=False)
+
+        if not self.inputs[-1].is_linked:
+            row = col.row(align=True)
+            for direction in "XYZ":
+                op = row.operator("node.set_sort_direction", text=direction)
+                op.direction = direction
+
+        col = layout.column(align=True)
+        row = col.row(align=True)
+        row.prop(self, "mode", expand=False, text="")
+        layout.prop(self, "descending")
+
+    def process(self):
+        if not any(s.is_linked for s in self.outputs):
+            return
+
+        inputs = self.inputs
+        input_v = inputs["Verts"].sv_get()
+        input_p = inputs["Polys"].sv_get()
+        input_r = inputs[-1].sv_get()[0]  # reference: direction or point
+
+        params = match_long_repeat([input_v, input_p, input_r])
+
+        iList, vList, pList, qList = [], [], [], []
+        for v, p, r in zip(*params):
+            indices, polys, quantities = self.sort_polygons(v, p, r)
+            iList.append(indices)
+            vList.append(v)
+            pList.append(polys)
+            qList.append(quantities)
+
+        if self.outputs['Vertices'].is_linked:
+            self.outputs['Vertices'].sv_set(vList)
+        if self.outputs['Polygons'].is_linked:
+            self.outputs['Polygons'].sv_set(pList)
+        if self.outputs['Indices'].is_linked:
+            self.outputs['Indices'].sv_set(iList)
+        if self.outputs[-1].is_linked:
+            self.outputs[-1].sv_set(qList) # sorting quantities
+
```

Comment:
consider: any data already calculated can be passed to the output, without testing if the socket's linked.   Dropping the `is_linked` tests makes the process function a little noisy.

Reply:
Doesn't it have any performance penalty to pass the data to an unconnected output?

Reply:
it's merely passing a reference, not an instance. - the real performance penalty will be dealt by calculating data for sockets that aren't connected in the first place.

Reply:
sometimes it makes sense to calculate everything in one pass, if there's reuse of calculated variables.

it's not a massive issue, but i wouldn't bother testing output `sockets.is_linked` most of the time - only if the processing step also tests if an output is connected prior to data-manipulation

team-sverchok has no explicit rules about this.

----
explicitely only passing data when a socket is_linked,  has prohibited us from doing cooler node composition stuff too (programatically chaining the process functions of nodes to compose new nodes on the fly --- topic for another time perhaps)